modules/javafx.graphics/src/main/native-glass/win/GlassApplication.cpp

```diff
@@ -181,20 +168,25 @@ LRESULT GlassApplication::WindowProc(UINT msg, WPARAM wParam, LPARAM lParam)
             }
             break;
         case WM_SETTINGCHANGE:
+            if (((UINT)wParam == SPI_GETHIGHCONTRAST ||
+                    lParam != NULL && wcscmp(LPCWSTR(lParam), L"ImmersiveColorSet") == 0) &&
+                    m_platformSupport.updatePreferences(m_grefThis)) {
+                return 0;
```

Comment:
Do we need to fall through in this case? We used to do so, which is why I'm asking.

Reply:
I decided not to fall through because the documentation for [WM_SETTINGCHANGE](https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-settingchange) says that applications should return zero if they process the message.